nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,61 @@
+const express = require("express");
+const router = express.Router();
+const meals = require("./../data/meals");
+const notANumber = { message: "Not an integer" };
+const errorMessage = { message: "Something wrong" };
+
+router.get("/:id", async(request, response) => {
+    const neededId = Number(request.params.id);
+    if (!Number.isInteger(needdedId)) {
+        return response.status(400).json(notANumber);
+    } else {
+        let newObj = {};
+        meals.find((element) => {
+            if (element.id === neededId) {
+                newObj = element;
+            }
+        });
+        return response.send(newObj);
+    }
+});
+
+// I do not know how to return all meals in my case.
+
+router.get("/", async(request, response) => {
+    const maxPrice = Number(request.query.maxPrice);
+    const title = request.query.title;
+    const createdAfter = new Date(request.query.createdAfter);
+    const limit = Number(request.query.limit);
+    if (maxPrice) {
```

Comment:
In case maxPrice is NaN the condition of this 'if' would be false so checking for isNaN again is not necessary